### PR TITLE
Add Devise gem for user authentication

### PR DESCRIPTION
**Pull request by jestradadeveloper**
# Description
Add devise gem, generation of sign in, sign out, recovery and all devise views
creation of user devise routes and migration:
 
  ## Screenshots 
Captura de Pantalla 2021-10-04 a la(s) 17 58 37
Captura de Pantalla 2021-10-04 a la(s) 17 59 17

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -4,3 +4,5 @@ DATABASE_PW = database-pasword
 DATABASE_HOST = database-host
 DATABASE_PORT = database-port
 DATABASE_TEST = database-name-test
+MAILER_DEFAULT_URL_OPTIONS_HOST = MAILER_DEFAULT_URL_OPTIONS_HOST
+MAILER_DEFAULT_URL_OPTIONS_PORT = MAILER_DEFAULT_URL_OPTIONS_PORT
```
Comment (OswaldoPineda) [inline .env.example:8]: add an extra line here

**Review by OswaldoPineda — APPROVED** [review]

**Review by YilKanda — APPROVED** [review]
LGTM!

File: .env.example (hunk 1 above)
Comment (JuanequeX) [reply .env.example:8]: Thank you for reviewing us, we will apply the changes immediately.

Comment (jestradadeveloper) [reply .env.example:8]: @OswaldoPineda  Should i put this line before MAILER_DEFAULT_URL_OPTIONS_HOST  or after both of the new lines added?

**Review by xFerchoVJ — DISMISSED** [review]
LGTM!

**Review by OswaldoPineda — APPROVED** [review]

**Review by YilKanda — APPROVED** [review]
LGTM!

**Pull request merged**